### ISSUE TITLE
Error trying to adding an alias to an existing Subscription

### ISSUE THREAD

**Issue by JoelCanteroGCO** [hashicorp/terraform-provider-azurerm#23014]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.5

### AzureRM Provider Version

3.69.0

### Affected Resource(s)/Data Source(s)

azurerm_subscription

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
  client_id     = ""
  client_secret = ""
  tenant_id     = ""
  subscription_id = ""
}

data "azurerm_subscriptions" "available" {
}

locals {
  subscriptions_data = csvdecode(file("${path.module}/subscription_data.csv"))
  subscriptions_map = {
    for obj in data.azurerm_subscriptions.available.subscriptions.* : obj.display_name => obj
  }
}

data "azurerm_subscription" "sub" {
  for_each = local.subscriptions_map

  subscription_id     = each.value.subscription_id
}

resource "azurerm_subscription" "subscription" {
  for_each = { for subscription in local.subscriptions_data : subscription.subscription_name => subscription }
  alias = data.azurerm_subscription.sub[each.key].display_name
  subscription_name = data.azurerm_subscription.sub[each.key].display_name
  subscription_id = data.azurerm_subscription.sub[each.key].subscription_id
}
```

### Debug Output/Panic Output

```shell
Error: checking for existence of Subscription by Alias "ds-sin013-dev": subscriptions.SubscriptionsClient#AliasGet: Failure responding to request: StatusCode=401 -- Original Error: autorest/azure: Service returned an error. Status=401 Code="UserNotAuthorized" Message="User does not have access Microsoft.Subscription/aliases/write over scope providers/Microsoft.Subscription/aliases/ds-sin013-dev"
│
│   with azurerm_subscription.subscription["ds-sin013-dev"],
│   on main.tf line 30, in resource "azurerm_subscription" "subscription":
│   30: resource "azurerm_subscription" "subscription" {
```

### Expected Behaviour

Trying to follow this [article](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/subscription#example-usage---adding-an-alias-to-an-existing-subscription) (adding an Alias to an existing Subscription), it returns this error and it is strange, because the Service Principal of the provider is owner in this subscription and in all management group root.

I am trying to iterate through a CSV where I have all the tag values that I would like to assign to each subscription. 

### Actual Behaviour

It throws this error:

Error: checking for existence of Subscription by Alias "ds-sin013-dev": subscriptions.SubscriptionsClient#AliasGet: Failure responding to request: StatusCode=401 -- Original Error: autorest/azure: Service returned an error. Status=401 Code="UserNotAuthorized" Message="User does not have access Microsoft.Subscription/aliases/write over scope providers/Microsoft.Subscription/aliases/ds-sin013-dev"

### Steps to Reproduce

terraform apply -auto-approve

**Comment by IanTurtle** [comment]
Any luck on this? Experiencing the same issue.

**Comment by JoelCanteroGCO** [comment]
No, same problem and the newer versions does not fix this error.

**Comment by IanTurtle** [comment]
After some digging around - I noticed we were trying to create a new subscription, with the same name as an already existing subscription. Maybe check to see you're not creating a duplicate.

**Comment by thevman** [comment]
Hey guys, I'm facing the same issue as well.

**Comment by eehret** [comment]
Still happening as of 3.75.0

**Comment by JoelCanteroGCO** [comment]
Hi, any news about this?

Thank you,
Joel

**Comment by asos-tomswaby** [comment]
Also seeing this problem. How can you reliably import a subscription into state?

**Comment by Fodsuk** [comment]
this issue seems to be related to the alias API (https://learn.microsoft.com/en-us/rest/api/subscription/alias/get?view=rest-subscription-2021-10-01&tabs=HTTP)

It seems calling this endpoint with an alias that doesn't exist, BUT does match an existing subscription name returns the error. 

```
existing subscription name: mike terraform subscription.
calling **providers/Microsoft.Subscription/aliases/mike-terraform-subscription?api-version=2021-10-01** returns the error
```

**Comment by Poltergeisen** [comment]
I'm having this issue but can't figure out where to assign these permissions. Is being an owner/contributor at the management group scope not good enough for this?

**Comment by gthomson31** [comment]
Same issue here - did anyone find a fix ? 

Example 

Subscription Create : 
```hcl
resource "azurerm_subscription" "sub" {
  provider = azurerm.tenant1
  subscription_name = "Example"
  billing_scope_id = "ABCDEFG"
}

```

Provider Service Principal
```hcl
provider "azurerm" {
  features {}
  alias           = "tenant1"
  client_id       = <REDACTED>
  client_secret   = <REDACTED>
  tenant_id       = local.tenant1
  subscription_id = "example-sub-id"
}
```
The service principal has been setup with owner permissions inside the relevant subscription but is still unable to run a plan

**Comment by gthomson31** [comment]
Interesting - having a similar issue imported subscriptions into a state and created under my account now trying to shift them to SP and getting thrown the same error.

- SP is Owner in Invoicing / Billing Profile / Subscription 


Cant think of what else !
